src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -225,7 +225,17 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
                 }
                 else
                 {
-                    DicomElement de = (DicomElement)item;
+                    DicomElement de;
```

Comment:
I am curious about this whole method. We are validating all items now, why? I though we only validate things that we index or during json conversion.